### PR TITLE
Update grpc to v1.57.2

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-grpc-server/build.gradle.kts [hunk 1]
```diff
@@ -21,7 +22,13 @@ dependencies {
   implementation(libs.grpc.stub) {
     exclude(group = "com.google.guava", module = "guava")
   }
-  implementation("com.google.guava:guava:21.0")
+  implementation(libs.checker.qual)
+  implementation(libs.guava){
+    attributes {
+      // We help Gradle pick between the jre and android of Guava.
```
Comment (Egorand) [inline wire-grpc-server/build.gradle.kts:28]: Is it needed? Does Gradle resolve it to the Android version by default?

**Review by Egorand — APPROVED** [review]

File: wire-grpc-server/build.gradle.kts (hunk 1 above)
Comment (oldergod) [reply wire-grpc-server/build.gradle.kts:28]: It doesn't resolve it at all actually so I had to tell him

image

And others are seeing it: https://github.com/google/guava/issues/6612

Comment (oldergod) [reply wire-grpc-server/build.gradle.kts]: I talked with @martinbonnin and we cannot repro without the attribute so I tried a commit without any of them.

Comment (martinbonnin) [reply wire-grpc-server/build.gradle.kts]: 🤞

**Pull request merged**